app/jenkins-for-jira-ui/src/components/ConnectJenkins/CreateServer/CreateServer.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from 'react';
+import Button, { LoadingButton } from '@atlaskit/button';
+import Form, {
+	FormFooter
+} from '@atlaskit/form';
+import { v4 as uuidv4 } from 'uuid';
+import { useHistory } from 'react-router';
+import {
+	loadingIcon
+} from '../../JenkinsConfigurationForm/JenkinsConfigurationForm.styles';
+import { createJenkinsServer } from '../../../api/createJenkinsServer';
+import { ConfigurationSteps } from '../ConfigurationSteps/ConfigurationSteps';
+import { StyledH1, StyledInstallationContainer, StyledInstallationContent } from '../ConnectJenkins.styles';
+import { generateNewSecret } from '../../JenkinsConfigurationForm/JenkinsConfigurationForm';
+import { isFormValid, setName } from '../../../common/util/jenkinsConnectionsUtils';
+import { ServerConfigurationFormName } from '../../JenkinsConfigurationForm/ServerConfigurationFormElements/ServerConfigurationFormName/ServerConfigurationFormName';
+import { ConnectLogos } from '../ConnectLogos/ConnectLogos';
+
+const CreateServer = () => {
+	const history = useHistory();
+	const [uuid] = useState(uuidv4);
+	const [serverName, setServerName] = useState('');
+	const [secret, setSecret] = useState('');
+	const [hasError, setHasError] = useState(false);
+	const [isLoading, setIsLoading] = useState(false);
+	const [errorMessage, setErrorMessage] = useState('');
+
+	useEffect(() => {
+		setSecret(generateNewSecret());
+	}, [uuid]);
+
+	const createServer = async () => {
```

Comment:
I wouldn't call this function the same name as the component.  I tend to go with what it's reacting to, in this case `onSubmit`, but that's just personal preference.

Reply:
I agree that the function shouldn't have the same name as the component, but I tend to stay away from function names like `onSubmit` because it doesn't summarise what the function is doing. You have to read the code to understand what happens when `onSubmit` is called.

Another benefit of having a more specific function name is that it allows you to see what the onSubmit will do from the Form component tag, i.e. `<Form onSubmit={createServer}>`, rather than you having to scroll up to the `onSubmit` function for more information.

I could rename it to something like `submitCreateServer`. I'll come up with a better name and update this.

Reply:
The problem with that is you're not just creating the server, you're validating the form, then creating the server if valid, so your function name isn't describing what it's doing either.

Reply:
As discussed via Slack, I've updated the name to `onSubmitCreateServer` to match the standard practice of naming functions `onXyz` when they are intended to be used by event handlers.